packages/node/src/authenticatedFetch/fetchFactory.ts

```diff
@@ -22,33 +22,89 @@
 import { JWK, JWT } from "jose";
 import { fetch } from "cross-fetch";
 import { v4 } from "uuid";
+import { ITokenRefresher } from "../login/oidc/refresh/TokenRefresher";
 
 // node-fetch fetch has an additional property (isRedirect) which prevents using
 // `typeof fetch`.
 // export type fetchType = (info: RequestInfo, init?: RequestInit) => Promise<Response>;
 export type fetchType = typeof fetch;
 
+export type RefreshOptions = {
+  sessionId: string;
+  refreshToken: string;
+  tokenRefresher: ITokenRefresher;
+};
+
+function isExpectedAuthError(statusCode: number): boolean {
+  // As per https://tools.ietf.org/html/rfc7235#section-3.1 and https://tools.ietf.org/html/rfc7235#section-3.1,
+  // a response failing because the provided credentials aren't accepted by the
+  // server can get a 401 or a 403 response.
+  return [401, 403].includes(statusCode);
+}
+
 /**
- * @param authToken A bearer token.
- * @param _refreshToken An optional refresh token.
+ * @param accessToken A bearer token.
+ * @param refreshToken An optional refresh token.
  * @returns A fetch function that adds an Authorization header with the provided
  * bearer token.
  * @hidden
  */
 export function buildBearerFetch(
-  authToken: string,
-  // TODO: We need to push this refresh token into a wrapper around the fetch,
-  //  so dependent on that wrapper existing first!
-  _refreshToken: string | undefined
+  accessToken: string,
+  refreshOptions?: RefreshOptions
 ): fetchType {
-  return (init: RequestInfo, options?: RequestInit): Promise<Response> => {
-    return fetch(init, {
+  // currentAccessToken and currentRefreshOptions are initialized with the provided
```

Comment:
I'm always wary of explicitly naming things like variables or function/method names in comments, suggest:
```suggestion
  // These local variables are initialized with the provided input
```

Reply:
Resolved in bbba942939219f8c7351bee9bfa17ed20ee74dfe